runtime/array_functions.h

```diff
@@ -789,23 +789,31 @@ array<T> f$array_diff_key(const array<T> &a1, const array<T1> &a2) {
   return result;
 }
 
-template<class T, class T1>
-array<T> f$array_diff(const array<T> &a1, const array<T1> &a2) {
+template<typename T, typename T1, typename Proj>
+array<T> array_diff_impl(const array<T> &a1, const array<T1> &a2, const Proj &projector) {
   array<T> result(a1.size());
 
   array<int64_t> values(array_size(0, a2.count(), false));
```

Comment:
The second param here is about of the expected number of string keys in array (for the reserving). 
For int specialization it is better to use the first parameter for reserving space for the integer keys.

Reply:
Yea, nice idea. It allow to speed up my synthetic benchmark from ~1.6 to ~1.5 sec, but the code became a bit more complicated. I'm not sure worth this changes to be merged?

Reply:
this is fine, let's leave as is